Makefile

```diff
@@ -0,0 +1,33 @@
+.PHONY: all format
+
+# format rust code using the specified command for the specified file or directory.
+# $(call do-format-with-cmd,cmd,file-or-dir)
+define do-format-with-cmd
+$1 --write-mode diff $2 |  grep -E "Diff .*at line" > /dev/null && $1 --write-mode overwrite $2 || exit 0
+endef
+
+# format rust code in the specified file or directory
+# a file of rust code follows the convention of having suffix '.rs'
+# $(call format-code-in,file-or-dir)
+define format-code-in
+  $(if $(filter %.rs,$1),  \
+      $(QUEIT) $(call do-format-with-cmd,rustfmt,$1), \
```

Comment:
`$(QUEIT)` could be removed since `@` is used in line 27~29.
It's a good practice to use `$(QUEIT)` to be set to `@`. This variable `QUEIT` expends to `@` usually. And we could set `QUEIT` to `""` in command line or env variables when we need to debug what commands to be executed for a target.